gpt4all-bindings/python/gpt4all/gpt4all.py

```diff
@@ -219,42 +235,69 @@ def generate(
         )
 
         if self._is_chat_session_activated:
+            generate_kwargs['reset_context'] = len(self.current_chat_session) == 1 # check if there is only one message, i.e. system prompt
             self.current_chat_session.append({"role": "user", "content": prompt})
-            generate_kwargs['prompt'] = self._format_chat_prompt_template(messages=self.current_chat_session[-1:])
-            generate_kwargs['reset_context'] = len(self.current_chat_session) == 1
+            prompt = self._format_chat_prompt_template(
+                                messages = self.current_chat_session[-1:],
+                                default_prompt_header = self.current_chat_session[0]["content"] if generate_kwargs['reset_context'] else ""
+                            )
         else:
             generate_kwargs['reset_context'] = True
+        
+        output_collector: List[MessageList] = [{"content": ""}] # placeholder for the self.current_chat_session if chat session is not activated
```

Comment:
It's defined as `MessageType` not `MessageList`. Better in general would be (because you assign it later again):
```py
        output_collector: List[MessageType]
        output_collector = [{"content": ""}] # placeholder for the self.current_chat_session if chat session is not activated
```